### PR TITLE
Fix bug that use obj as prev obj in endpoint slice update

### LINKED ISSUE #92337: Mistake use obj as prevObj in onEndpointSliceUpdate
**What happened**:
Mistake use obj as prevObj， prevEndpointSlice := obj.(*discovery.EndpointSlice) 

https://github.com/kubernetes/kubernetes/blob/0a91893126e6becf60777d0f14ade79e4058bc06/pkg/controller/endpointslice/endpointslice_controller.go#L406-L416

**What you expected to happen**:

prevEndpointSlice := prevObj.(*discovery.EndpointSlice) 

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

cc @robscott

### PR DESCRIPTION
**Pull request by fatkun**
**What type of PR is this?**
/kind bug
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #92337

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
Fixed a bug that mistake use newObj as oldObj in endpoint slice update.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
```

### CODE REVIEW

File: pkg/controller/endpointslice/endpointslice_controller_test.go [hunk 1]
```diff
@@ -274,6 +274,31 @@ func TestSyncServiceEndpointSliceLabelSelection(t *testing.T) {
...
+		AddressType: discovery.AddressTypeIPv4,
+	}
+
+	epSlice2 := epSlice1.DeepCopy()
+	epSlice2.Labels[discovery.LabelManagedBy] = "something else"
+
+	assert.Equal(t, 0, esController.queue.Len())
+	esController.onEndpointSliceUpdate(epSlice1, epSlice2)
+	time.Sleep(1 * time.Second)
+	assert.Equal(t, 1, esController.queue.Len())
```
Comment (robscott) [inline pkg/controller/endpointslice/endpointslice_controller_test.go:299]: Unfortunately the test failures on this PR make it look like this delay will end up having flaky results. There may be a better more reliable way to see if something has been queued. One option might be to use `wait.PollImmediate()` to check every second for up to 3 seconds if something has been added to the queue and return true if it has.

**Review by robscott — COMMENTED** [review]
Thanks for finding and fixing this!

File: pkg/controller/endpointslice/endpointslice_controller_test.go (hunk 1 above)
Comment (fatkun) [reply pkg/controller/endpointslice/endpointslice_controller_test.go]: I change to use wait.PollImmediate() to check every 100ms for up to 3 seconds.

Comment (robscott) [reply pkg/controller/endpointslice/endpointslice_controller_test.go]: This looks good, thanks! It looks like you just need to run `hack/update-bazel.sh` to get the verify test to pass now. Also once you've done that do you mind rebasing and squashing this back into a single commit for the PR? Thanks for your work on this!

**Comment by fatkun** [comment]
@robscott The test all pass now.

**Comment by robscott** [comment]
Thanks! Assigning Minhan for final approval.

/lgtm
/assign @freehan

**Pull request merged**